stellargraph/mapper/full_batch_generators.py

```diff
@@ -503,3 +505,26 @@ def flow(self, node_ids, targets=None):
         return RelationalFullBatchNodeSequence(
             self.features, self.As, self.use_sparse, targets, node_indices
         )
+
+
+class CorruptedGenerator:
+    """
+    Keras compatible data generator that wraps :class: `FullBatchNodeGenerator` and provides corrupted
+    data for training Deep Graph Infomax.
+
+    Args:
+        base_generator: the uncorrupted Sequence object.
+    """
+
+    def __init__(self, base_generator):
+
+        if not isinstance(base_generator, FullBatchNodeGenerator,):
+            raise TypeError(
+                f"base_generator: expected FullBatchNodeGenerator, "
+                f"found {type(base_generator).__name__}"
+            )
+        self.base_generator = base_generator
+
+    def flow(self, *args, **kwargs):
```

Comment:
This should have docs that point the describe the connection between these args and the `base_generator` ones.